### ISSUE TITLE
Stop panicking when the metrics listener port is unavailable

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#671]
When we have multiple zebrad instances running, port listener conflicts can cause panics and errors.

Metrics panics:
```
The application panicked (crashed).
Message: error binding to 0.0.0.0:9999: error creating server listener: Address already in use (os error 98)
Location: ~/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.13.7/src/server/mod.rs:124
```

Tracing fails, but continues:
```
ERROR zebrad::components::tracing: Could not open tracing endpoint listener
ERROR zebrad::components::tracing: Error: error creating server listener: Address already in use (os error 98)
```
(Sometimes the second line isn't logged.)

One of these errors results in a double panic, but I'm not sure which one.

Follow up to #669.

**Comment by hdevalence** [comment]
I think that panicking (or returning an error that terminates the program) is a reasonable choice here, *as long as the metrics listener port was specified explicitly*.  If we change the metrics listener config setting to be an `Option<SocketAddr>` with default `None`, then I think we'd get that behavior (although we'd still need #669).

It does seem like a bug that the behavior is inconsistent between the metrics endpoint and the tracing endpoint, although the tracing endpoint should eventually be subsumed into a general RPC endpoint (while the metrics endpoint shouldn't, because it speaks prometheus).  So the inconsistency might not be worth fixing before we rewrite that code.

**Comment by teor2345** [comment]
I think this issue will be resolved by #660.
